common/src/main/java/com/wynntils/handlers/item/ItemHandler.java

```diff
@@ -104,13 +108,35 @@ private void onItemStackUpdate(ItemStack existingItem, ItemStack newItem) {
             return;
         }
 
+        // I don't actually know a clean way to get 2 returns from one method when one is a boolean for an if condition
+        // or somehow not repeat a lot of code and seperate it to 2 methods.
+        int[] shinyStatLineIndex = {-1};
+
         // This might be just a name update. Check if lore matches:
-        if (!LoreUtils.loreSoftMatches(existingItem, newItem, 3)) {
+        if (!LoreUtils.loreSoftMatches(existingItem, newItem, 3, shinyStatLineIndex)) {
             // This could be a new item, or a crafted item losing in durability
             annotate(newItem);
             return;
         }
 
+        if (shinyStatLineIndex[0] != -1) {
+            int index = shinyStatLineIndex[0];
+            List<StyledText> existingItemLore = LoreUtils.getLore(existingItem);
+            List<StyledText> newItemLore = LoreUtils.getLore(newItem);
+
+            StyledText existingLine = existingItemLore.get(index).getNormalized();
+            StyledText newLine = newItemLore.get(index).getNormalized();
+
+            Matcher existingLineMatcher = existingLine.getMatcher(SHINY_STAT_PATTERN);
+            String statisticName = existingLineMatcher.group(1);
+            int existingStatistic = Integer.parseInt(existingLineMatcher.group(2));
+
+            Matcher newLineMatcher = newLine.getMatcher(SHINY_STAT_PATTERN);
+            int newStatistic = Integer.parseInt(newLineMatcher.group(2));
+            WynntilsMod.postEvent(
+                    new ShinyStatisticUpdateEvent(newItem, statisticName, existingStatistic, newStatistic));
```

Comment:
Can you hold off on adding the event until it is actually used?

Reply:
ok :thumbs_up: